From 3150fc51ddc49fba28898f94b51d299252e794d5 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Mon, 22 Dec 2025 13:04:14 +0000
Subject: [PATCH] Revert "Forward declare SingleThreadTaskRunner and
 SequencedTaskRunner"

This reverts commit 75de249a447bb5933cb448dc0bca7a45ba3e809e.
---
 base/metrics/user_metrics.h     | 2 +-
 base/observer_list_threadsafe.h | 1 +
 base/one_shot_event.h           | 1 +
 base/task/current_thread.h      | 2 +-
 base/task/thread_pool.h         | 2 +-
 dbus/object_manager.cc          | 1 -
 dbus/object_proxy.h             | 5 +----
 7 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/base/metrics/user_metrics.h b/base/metrics/user_metrics.h
index 9e173954da..818155973a 100644
--- a/base/metrics/user_metrics.h
+++ b/base/metrics/user_metrics.h
@@ -10,10 +10,10 @@
 #include "base/base_export.h"
 #include "base/functional/callback.h"
 #include "base/metrics/user_metrics_action.h"
+#include "base/task/single_thread_task_runner.h"
 
 namespace base {
 
-class SingleThreadTaskRunner;
 class TimeTicks;
 
 // This module provides some helper functions for logging actions tracked by
diff --git a/base/observer_list_threadsafe.h b/base/observer_list_threadsafe.h
index 97ce165d29..f54f2ddb31 100644
--- a/base/observer_list_threadsafe.h
+++ b/base/observer_list_threadsafe.h
@@ -23,6 +23,7 @@
 #include "base/strings/strcat.h"
 #include "base/synchronization/lock.h"
 #include "base/task/sequenced_task_runner.h"
+#include "base/task/single_thread_task_runner.h"
 #include "build/build_config.h"
 
 ///////////////////////////////////////////////////////////////////////////////
diff --git a/base/one_shot_event.h b/base/one_shot_event.h
index c55ce4cf05..736c57fb49 100644
--- a/base/one_shot_event.h
+++ b/base/one_shot_event.h
@@ -12,6 +12,7 @@
 #include "base/functional/callback_forward.h"
 #include "base/sequence_checker.h"
 #include "base/task/sequenced_task_runner.h"
+#include "base/task/single_thread_task_runner.h"
 
 namespace base {
 
diff --git a/base/task/current_thread.h b/base/task/current_thread.h
index d84dd36760..d0615fee23 100644
--- a/base/task/current_thread.h
+++ b/base/task/current_thread.h
@@ -18,6 +18,7 @@
 #include "base/message_loop/message_pump_for_ui.h"
 #include "base/pending_task.h"
 #include "base/task/sequence_manager/task_time_observer.h"
+#include "base/task/single_thread_task_runner.h"
 #include "base/task/task_observer.h"
 #include "build/build_config.h"
 
@@ -36,7 +37,6 @@ class WebTaskEnvironment;
 namespace base {
 
 class CallbackListSubscription;
-class SingleThreadTaskRunner;
 
 namespace test {
 bool RunUntil(FunctionRef<bool(void)>);
diff --git a/base/task/thread_pool.h b/base/task/thread_pool.h
index cc3649a9e2..9ae560f822 100644
--- a/base/task/thread_pool.h
+++ b/base/task/thread_pool.h
@@ -16,6 +16,7 @@
 #include "base/location.h"
 #include "base/memory/scoped_refptr.h"
 #include "base/task/post_task_and_reply_with_result_internal.h"
+#include "base/task/single_thread_task_runner.h"
 #include "base/task/single_thread_task_runner_thread_mode.h"
 #include "base/task/task_runner.h"
 #include "base/task/task_traits.h"
@@ -26,7 +27,6 @@
 namespace base {
 
 class SequencedTaskRunner;
-class SingleThreadTaskRunner;
 
 // This is the interface to post tasks to base's thread pool.
 //
diff --git a/dbus/object_manager.cc b/dbus/object_manager.cc
index 0d3ebfbe65..93e6c87ee6 100644
--- a/dbus/object_manager.cc
+++ b/dbus/object_manager.cc
@@ -12,7 +12,6 @@
 #include "base/logging.h"
 #include "base/metrics/histogram.h"
 #include "base/strings/stringprintf.h"
-#include "base/task/sequenced_task_runner.h"
 #include "dbus/bus.h"
 #include "dbus/dbus_statistics.h"
 #include "dbus/error.h"
diff --git a/dbus/object_proxy.h b/dbus/object_proxy.h
index dc5376449d..18b1ca0e8c 100644
--- a/dbus/object_proxy.h
+++ b/dbus/object_proxy.h
@@ -17,16 +17,13 @@
 #include "base/functional/callback.h"
 #include "base/memory/raw_ptr.h"
 #include "base/memory/ref_counted.h"
+#include "base/task/sequenced_task_runner.h"
 #include "base/time/time.h"
 #include "base/types/expected.h"
 #include "dbus/dbus_export.h"
 #include "dbus/error.h"
 #include "dbus/object_path.h"
 
-namespace base {
-class SequencedTaskRunner;
-}
-
 namespace dbus {
 
 class Bus;
-- 
2.52.0.322.g1dd061c0dc-goog

